dependency.lic

```diff
@@ -770,8 +770,34 @@ class SetupFiles
     boxes = @base_files['base-items.yaml'][:box_nouns]
     gems = @base_files['base-items.yaml'][:gem_nouns]
     scrolls = @base_files['base-items.yaml'][:scroll_nouns]
-    s.lootables = (base_lootables + boxes + gems + scrolls + s.loot_additions - s.loot_subtractions).uniq
+    spells = @base_files['base-spells.yaml'][:spell_data]
 
+    s.lootables = (base_lootables + boxes + gems + scrolls + s.loot_additions - s.loot_subtractions).uniq
+    s.offensive_spells
+      .select { |spell| spells[spell['name']] }
+      .each_with_index do |spell, i|
+        s.offensive_spells[i] = spells[spell['name']].merge(spell)
+      end
+    s.buff_spells
+      .select { |spell| spells[spell] }
+      .each do |name, _data|
+        s.buff_spells[name] = spells[name].merge(s.buff_spells[name])
+      end
+    s.lockpick_buffs['spells']
+      .each_with_index
+      .select do |spell, i| 
+        spells.select { |_name, data| data['abbrev'].downcase == spell['abbrev'].downcase }
+              .each { |_name, data| s.lockpick_buffs['spells'][i] = data.merge(spell) } 
+      end
+    s.astrology_buffs['spells']
+      .each_with_index
```

Comment:
This was a googled solution to 'select with index ruby'. I wanted to retain the index of what select is returning